src/libra_py/dynamics/tsh/save.py

```diff
@@ -473,13 +481,15 @@ def save_hdf5_2D(saver, i, states, txt_type=0):
 
 
 
-def save_hdf5_2D_new(saver, i, dyn_var, ham, txt_type=0):
+def save_hdf5_2D_new(saver, i, params, dyn_var, ham, txt_type=0):

```

Comment:
please remove the `params` argument from the inputs? the idea of the saver functions it to save things based on what is requested to be saved - only.  Not do decision making - this brings the design away from the modularity. It is okay to not have the proper variables (then just zeroes will be saved). The computation of needed properties should be done elsewhere, except for the single function that uses the pre-computed properties into the desired one.  
Also, if you make the saving of properties conditions - depending on the `rep_sh` parameter you lose flexibility in printing anything. One way around is to just have two kinds of diabatic SH populations - based on Tempelaar and on the diabatic trajectories - these would just be certain keywords, which can be called or not (e.g. we don't expect anything there but zeroes)